app/src/main/java/com/sopt/now/ui/LoginActivity.kt

```diff
@@ -0,0 +1,78 @@
+package com.sopt.now.ui
+
+import android.content.Intent
+import android.os.Bundle
+import android.widget.Toast
+import androidx.activity.result.ActivityResultLauncher
+import androidx.activity.result.contract.ActivityResultContracts
+import com.sopt.now.R
+import com.sopt.now.databinding.ActivityLoginBinding
+
+import com.sopt.now.util.BindingActivity
+
+class LoginActivity : BindingActivity<ActivityLoginBinding>(R.layout.activity_login) {
+    private lateinit var resultLauncher: ActivityResultLauncher<Intent>
+    private var ID = ""
+    private var password = ""
+    private var MBTI = ""
+    private var nickname = ""
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        init()
+    }
+
+    private fun init() {
+        setResultNext()
+        initButton()
+    }
+
+    private fun initButton() {
+        binding.btnLoginToSignup.setOnClickListener {
+            val intent = Intent(this, SignUpActivity::class.java)
+            resultLauncher.launch(intent)
+        }
+
+        binding.btnLogin.setOnClickListener {
+            if (validateLogin()) {
+                val intent = Intent(this, MainActivity::class.java)
+                intent.putExtra("ID", ID)
+                intent.putExtra("password", password)
+                intent.putExtra("nickname", nickname)
+                intent.putExtra("MBTI", MBTI)
+                startActivity(intent)
```

Comment:
```suggestion
    val intent = Intent(this, MainActivity::class.java).apply {
    putExtra("ID", ID)
    putExtra("password", password)
    putExtra("nickname", nickname)
    putExtra("MBTI", MBTI)
}
startActivity(intent)

```
이런식으로 scope함수로 묶어보는 건 어떨까요??

Reply:
변수별로 따로 전달을 해야할까요?? 객체로 전달하는 방법은 없을까요?

Reply:
scope 함수로 묶는 것도 처리했습니다! 
덕분에 @Parcelize 을  사용해서 해결했습니다.
과정속에서 Serializable ,Parcelable에 대해서도 알게되었구요. 
조만간 자료 정리해야겠어요
참고했던 자료는 아래와 같아요
[각각 용어에 대해](https://studyroadmap-kkm.tistory.com/167#Serializable-1)
[구현 엿보기](https://velog.io/@simsubeen/Kotlin-Parcelize-%EC%82%AC%EC%9A%A9%ED%95%98%EC%97%AC-intent%EC%97%90-data-class-%EC%A0%84%EB%8B%AC%ED%95%98%EA%B8%B0#detailactivity)

Reply:
login 이랑  singup쪽도 적용 가능할거 같은데 밤을 새서 그런가 머리가 안굴러가네요.. 다녀와서 해보겠습니다!